pkg/obfuscator/domain.go

```diff
@@ -2,102 +2,77 @@ package obfuscator
 
 import (
 	"fmt"
-	"path/filepath"
 	"regexp"
 	"strings"
 )
 
+const (
+	domainPattern      = `([a-zA-Z0-9\.]*\.)?(%s)`
+	obfuscatedTemplate = "domain%07d"
+)
+
 type domainObfuscator struct {
 	ReplacementReporter
-	hostPattern *regexp.Regexp
-	domainCount int
-	tlds        map[string]struct{}
-	mainDomains map[string]string
-	ipPattern   *regexp.Regexp
+	domainCount    int
+	domainPatterns []*regexp.Regexp
+	domainMapping  map[string]string
 }
 
 func (d *domainObfuscator) FileName(s string) string {
-	// if the filename has an extension omit it from obfuscation
-	// this is still error-prone because the last part of the domain name could still be mistaken for an extension
-	extension := filepath.Ext(s)
-	if extension != "" {
-		return fmt.Sprintf("%s%s", d.findDomains(s[:len(s)-len(extension)]), extension)
-	}
 	return d.findDomains(s)
 }
 
-func (d *domainObfuscator) findDomains(input string) string {
-	domains := d.hostPattern.FindAllString(input, -1)
+func (d *domainObfuscator) Contents(s string) string {
+	return d.findDomains(s)
+}
 
+func (d *domainObfuscator) findDomains(input string) string {
```

Comment:
does the naming still make sense? it replaces the domains it found

Reply:
Changed to `replaceDomains`.